### ISSUE TITLE
Dropdown does not do the thing that it should do and change its size massively.

### ISSUE THREAD

**Issue by applicatlat** [flet-dev/flet#1350]
**Description**

The problem is that when dropdown button is pressed and an option of 3 is selected, randomly, the dropdown button shape is changed. Additionally, time to time, the dropdown does not trigger the action always as it should. To express it more clearly, I took a video of the problem. This is the first time I encountered a problem like this. Within my project, there are two .py files. One has the page and its properties other has the class that the object is called.

**Code example to reproduce the issue:** 
There are no errors within the code

```python
deneme.py:

import flet as ft
import tres


def main(page: ft.Page):
    page.scroll = ft.ScrollMode.ALWAYS
    page.vertical_alignment = ft.CrossAxisAlignment.CENTER
    page.horizontal_alignment = ft.CrossAxisAlignment.CENTER

    def page_setter(lang):
        if int(page.width) < 2 * 310:
            page.add(tres.AjandaApp(language=lang, mode="one"))
            return
        if 2 * 310 <= int(page.width) < 3 * 310:
            page.add(tres.AjandaApp(language=lang, mode="two"))
            return
        if 3 * 310 <= int(page.width) < 4 * 310:
            page.add(tres.AjandaApp(language=lang, mode="three"))
            return
        if 4 * 310 <= int(page.width):
            page.add(tres.AjandaApp(language=lang, mode="four"))
            return

    def page_resize(e,lang):
        page.clean()
        page_setter(lang=lang)

    page_setter(lang=1)
    page.on_resize = lambda e,lang=1: page_resize(e,lang)


ft.app(target=main)

tres.py:
import flet as ft

lang_dict = {
    "months": [["01-Jan", "02-Feb", "03-Mar", "04-Apr", "05-May", "06-Jun", "07-Jul",
                "08-Aug", "09-Sep", "10-Oct", "11-Nov", "12-Dec"], ["01-Oca",
                                                                    "02-Şub", "03-Mar", "04-Nis", "05-May", "06-Haz",
                                                                    "07-Tem", "08-Ağu",
                                                                    "09-Eyl", "10-Ekİ", "11-Kas", "12-Ara"]],
    "day_written": [['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
                    ['Pazartesi', 'Salı', 'Çarşamba', 'Perşembe', 'Cuma', 'Cumartesi', 'Pazar']],
    "month_long": [['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October',
                    'November', 'December'], ['Ocak', 'Şubat', 'Mart', 'Nisan', 'Mayıs', 'Haziran', 'Temmuz', 'Ağustos',
                                              'Eylül', 'Ekim', 'Kasım', 'Aralık']],
    "ok": ["Got it.", "Tamamdır."],
    "month": ["MM", "AA"],
    "save": ["Save", "Kaydet"],
    "day": ["DD", "GG"],
    "year": ["YYYY", "YYYY"],
    "date_error": ["End date is bigger than the start date. Please re-check the values that you have typed and try to "
                   "save again.", "Bitiş tarihi, başlangıç tarihinden daha büyük. Lütfen girdiğiniz değerleri kontrol "
                                  "edin ve tekrar kaydetmeyi deneyin."],
    "start_date": ["Starting Date", "Başlangıç Tarihi"],
    "end_date": ["Ending Date (Non-mandatory)", "Bitiş Tarihi (Zorunlu değil)"],
    "pick_list": [["Medicine Planner", "Studying", "Exercise", "Single Event", "Re-occuring Event", "Custom"],
                  ["İlaç Düzenleyicisi", "Çalışma", "Egzersiz", "Tek Etkinlik", "Tekrarlanan Etkinlik", "Özel Ayar"]],
    "hm_days": ["For How Many Days?", "Kaç Gün Boyunca?"],
    "times_day": ["... Times a Day", "Günde ... Kere"],
    "pills": ["Total Number of Pills", "Toplam Hap/Doz Adedi"],
    "medicine_name": ["Name of the Medicine", "İlacın Adı"],
    "medicine_detail": ["Details Regarding Medicine", "İlaç Detayları"],
    "starting_hour": ["Starting Hour", "Başlangıç Saati"],
    "ending_hour": ["Ending Hour (Non-mandatory)", "Bitiş Saati (Zorunlu değil)"],
    "hour": ["Hour", "Saat"],
    "minute": ["Min", "Dak"],
    "title": ["Title", "Başlık"],
    "title_new": ["Please enter a new task title...", "Lütfen yeni bir görev başlığı girin..."],
    "hint_explanation": ["Please describe the task (non mandatory)...",
                         "Lütfen yapmak istediğiniz görevi detaylandırın (zorunlu değil)..."],
    "title_explanation": ["Describe Task", "Görevi Tanımlayın"],
    "every_day": ["Every ... Days", "... Günde Bir"],
    "every_hour": ["Every ... Hours", "Her ... Saatte Bir"],
    "every_month": ["Every ... Day of the Month", "Her Ayın ... Günü"],
    "date_title": ["DATE TAB", "TARİH TABI"],
    "medical_title": ["MEDICAL ARRANGEMENT TAB", "MEDİCAL DÜZENLEME TABI"],
    "picker_title": ["What would you like to organize?", "Ne düzenlemek isterdiniz?"],
    "enter_link": ["Enter URL", "Link Girin"],
    "url_pick": ["Pick Field to Enter URL", "Link Gireceğiniz Alanı Seçin"],
    "am": ["AM", "ÖÖ"],
    "pm": ["PM", "ÖS"],
    "sys": ["SYS", "SİS"],
}

minutes_list = [
    ft.dropdown.Option("00"), ft.dropdown.Option("01"), ft.dropdown.Option("02"), ft.dropdown.Option("03"),
    ft.dropdown.Option("04"), ft.dropdown.Option("05"), ft.dropdown.Option("06"), ft.dropdown.Option("07"),
    ft.dropdown.Option("08"), ft.dropdown.Option("09"), ft.dropdown.Option("10"), ft.dropdown.Option("11"),
    ft.dropdown.Option("12"), ft.dropdown.Option("13"), ft.dropdown.Option("14"), ft.dropdown.Option("15"),
    ft.dropdown.Option("16"), ft.dropdown.Option("17"), ft.dropdown.Option("18"), ft.dropdown.Option("19"),
    ft.dropdown.Option("20"), ft.dropdown.Option("21"), ft.dropdown.Option("22"), ft.dropdown.Option("23"),
    ft.dropdown.Option("24"), ft.dropdown.Option("25"), ft.dropdown.Option("26"), ft.dropdown.Option("27"),
    ft.dropdown.Option("28"), ft.dropdown.Option("29"), ft.dropdown.Option("30"), ft.dropdown.Option("31"),
    ft.dropdown.Option("32"), ft.dropdown.Option("33"), ft.dropdown.Option("34"), ft.dropdown.Option("35"),
    ft.dropdown.Option("36"), ft.dropdown.Option("37"), ft.dropdown.Option("38"), ft.dropdown.Option("39"),
    ft.dropdown.Option("40"), ft.dropdown.Option("41"), ft.dropdown.Option("42"), ft.dropdown.Option("43"),
    ft.dropdown.Option("44"), ft.dropdown.Option("45"), ft.dropdown.Option("46"), ft.dropdown.Option("47"),
    ft.dropdown.Option("48"), ft.dropdown.Option("49"), ft.dropdown.Option("50"), ft.dropdown.Option("51"),
    ft.dropdown.Option("52"), ft.dropdown.Option("53"), ft.dropdown.Option("54"), ft.dropdown.Option("55"),
    ft.dropdown.Option("56"), ft.dropdown.Option("57"), ft.dropdown.Option("58"), ft.dropdown.Option("59")
]

am_hours_list = [
    ft.dropdown.Option("01"), ft.dropdown.Option("02"), ft.dropdown.Option("03"), ft.dropdown.Option("04"),
    ft.dropdown.Option("05"), ft.dropdown.Option("06"), ft.dropdown.Option("07"), ft.dropdown.Option("08"),
    ft.dropdown.Option("09"), ft.dropdown.Option("10"), ft.dropdown.Option("11"), ft.dropdown.Option("12")
]

mil_hours_list = [
    ft.dropdown.Option("00"), ft.dropdown.Option("01"), ft.dropdown.Option("02"), ft.dropdown.Option("03"),
    ft.dropdown.Option("04"), ft.dropdown.Option("05"), ft.dropdown.Option("06"), ft.dropdown.Option("07"),
    ft.dropdown.Option("08"), ft.dropdown.Option("09"), ft.dropdown.Option("10"), ft.dropdown.Option("11"),
    ft.dropdown.Option("12"), ft.dropdown.Option("13"), ft.dropdown.Option("14"), ft.dropdown.Option("15"),
    ft.dropdown.Option("16"), ft.dropdown.Option("17"), ft.dropdown.Option("18"), ft.dropdown.Option("19"),
    ft.dropdown.Option("20"), ft.dropdown.Option("21"), ft.dropdown.Option("22"), ft.dropdown.Option("23")
]

icon_width = 40
drop_down_2d = 80
max_min_container_width = 310
fixed_margin = 20
inner_container_padding = 10
outer_container_padding = 10
border_radius = 20
title_size = 16
add_icon = ft.icons.CHECK
edit_icon = ft.icons.EDIT
delete_icon = ft.icons.DELETE
clear_icon = ft.icons.CLEAR
link_icon = ft.icons.LINK
warning_icon = ft.icons.TIPS_AND_UPDATES_ROUNDED


class AjandaApp(ft.UserControl):
    def __init__(self, language, mode):
        super().__init__()
        self.lang = language
        self.mode = mode
        self.am_pm_list = [
            ft.dropdown.Option(str(lang_dict["am"][self.lang])), ft.dropdown.Option(str(lang_dict["pm"][self.lang])),
            ft.dropdown.Option("24")
        ]
        self.new_task_title = ft.TextField(multiline=False, label=str(lang_dict["title"][self.lang]),
                                           border=ft.InputBorder.UNDERLINE,
                                           hint_text=str(lang_dict["title_new"][self.lang]))
        self.new_task_explanation = ft.TextField(multiline=True, label=str(lang_dict["title_explanation"][self.lang]),
                                                 border=ft.InputBorder.UNDERLINE,
                                                 hint_text=str(lang_dict["hint_explanation"][self.lang]))

        self.text_start_time = ft.Text(value=str(lang_dict["starting_hour"][self.lang]), size=title_size)
        self.text_end_time = ft.Text(value=str(lang_dict["ending_hour"][self.lang]), size=title_size)
        self.text_start_date = ft.Text(value=str(lang_dict["start_date"][self.lang]), size=title_size)
        self.text_end_date = ft.Text(value=str(lang_dict["end_date"][self.lang]), size=title_size)

        self.new_task_delete_ib = ft.IconButton(delete_icon, width=icon_width, data="delete")
        self.new_task_edit_ib = ft.IconButton(edit_icon, width=icon_width, data="edit")
        self.new_task_clear_ib = ft.IconButton(clear_icon, width=icon_width, data="clear")
        self.new_task_add_ib = ft.IconButton(add_icon, width=icon_width, data="add")
        self.new_task_link_ib = ft.IconButton(link_icon, width=icon_width, data="link")
        self.new_task_tooltip_ib = ft.IconButton(warning_icon, width=icon_width, data="tooltip")

        self.new_task_hour_system_end = ft.Dropdown(options=self.am_pm_list, data="am_pm_end", width=drop_down_2d,
                                                    alignment=ft.alignment.center, on_change=self.hour_controller_end,
                                                    label=str(lang_dict["sys"][self.lang]))
        self.new_task_minute_end = ft.Dropdown(options=minutes_list, width=drop_down_2d, alignment=ft.alignment.center,
                                               label=str(lang_dict["minute"][self.lang]), disabled=True)
        self.new_task_hour_end = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                             label=str(lang_dict["hour"][self.lang]), disabled=True)

        self.new_task_hour_system_start = ft.Dropdown(options=self.am_pm_list, data="am_pm_start", width=drop_down_2d,
                                                      alignment=ft.alignment.center,
                                                      on_change=self.hour_controller_start,
                                                      label=str(lang_dict["sys"][self.lang]))
        self.new_task_minute_start = ft.Dropdown(options=minutes_list, width=drop_down_2d,
                                                 alignment=ft.alignment.center,
                                                 label=str(lang_dict["minute"][self.lang]), disabled=True)
        self.new_task_hour_start = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                               label=str(lang_dict["hour"][self.lang]), disabled=True)

        self.new_task_day_start = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                              label=str(lang_dict["day"][self.lang]))
        self.new_task_month_start = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                                label=str(lang_dict["month"][self.lang]))
        self.new_task_year_start = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                               label=str(lang_dict["year"][self.lang]))

        self.new_task_day_end = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                            label=str(lang_dict["day"][self.lang]))
        self.new_task_month_end = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                              label=str(lang_dict["month"][self.lang]))
        self.new_task_year_end = ft.Dropdown(width=drop_down_2d, alignment=ft.alignment.center,
                                             label=str(lang_dict["year"][self.lang]))

        self.checkbox_hour_start = ft.Checkbox(data="1", on_change=self.check_box_checker)
        self.checkbox_hour_end = ft.Checkbox(data="2", on_change=self.check_box_checker)
        self.checkbox_date_start = ft.Checkbox(data="3", on_change=self.check_box_checker)
        self.checkbox_date_end = ft.Checkbox(data="4", on_change=self.check_box_checker)

        self.icon_text_column = ft.Column(controls=[self.new_task_link_ib, self.new_task_tooltip_ib],
                                          alignment=ft.alignment.center,
                                          horizontal_alignment=ft.CrossAxisAlignment.CENTER)

        self.row_0_controls = [self.new_task_clear_ib, self.new_task_add_ib, self.new_task_edit_ib,
                               self.new_task_delete_ib]
        self.row_1_controls = [self.new_task_title]
        self.row_2_controls = [self.checkbox_hour_start, self.text_start_time]
        self.row_3_controls = [self.new_task_hour_system_start, self.new_task_hour_start, self.new_task_minute_start]
        self.row_4_controls = [self.checkbox_hour_end, self.text_end_time]
        self.row_5_controls = [self.new_task_hour_system_end, self.new_task_hour_end, self.new_task_minute_end]
        self.row_6_controls = [self.checkbox_date_start, self.text_start_date]
        self.row_7_controls = [self.new_task_day_start, self.new_task_month_start, self.new_task_year_start]
        self.row_8_controls = [self.checkbox_date_end, self.text_end_date]
        self.row_9_controls = [self.new_task_day_end, self.new_task_month_end, self.new_task_year_end]
        self.row_10_controls = [self.icon_text_column, self.new_task_explanation]

        self.row0 = ft.Row(controls=self.row_0_controls, alignment=ft.MainAxisAlignment.SPACE_BETWEEN)
        self.row1 = ft.Row(controls=self.row_1_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row2 = ft.Row(controls=self.row_2_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row3 = ft.Row(controls=self.row_3_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row4 = ft.Row(controls=self.row_4_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row5 = ft.Row(controls=self.row_5_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row6 = ft.Row(controls=self.row_6_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row7 = ft.Row(controls=self.row_7_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row8 = ft.Row(controls=self.row_8_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row9 = ft.Row(controls=self.row_9_controls, alignment=ft.alignment.center,
                           vertical_alignment=ft.CrossAxisAlignment.CENTER)
        self.row10 = ft.Row(controls=self.row_10_controls, alignment=ft.alignment.center,
                            vertical_alignment=ft.CrossAxisAlignment.CENTER)

        self.column1 = ft.Column(controls=[self.row2, self.row3], alignment=ft.alignment.center,
                                 horizontal_alignment=ft.CrossAxisAlignment.CENTER)
        self.column2 = ft.Column(controls=[self.row4, self.row5], alignment=ft.alignment.center,
                                 horizontal_alignment=ft.CrossAxisAlignment.CENTER)
        self.column3 = ft.Column(controls=[self.row6, self.row7], alignment=ft.alignment.center,
                                 horizontal_alignment=ft.CrossAxisAlignment.CENTER)
        self.column4 = ft.Column(controls=[self.row8, self.row9], alignment=ft.alignment.center,
                                 horizontal_alignment=ft.CrossAxisAlignment.CENTER)

        self.cont1 = ft.Container(self.column1, alignment=ft.alignment.center, border_radius=border_radius,
                                  border=ft.border.all(2, ft.colors.BLACK), padding=inner_container_padding,
                                  width=max_min_container_width - 28)
        self.cont2 = ft.Container(self.column2, alignment=ft.alignment.center, border_radius=border_radius,
                                  border=ft.border.all(2, ft.colors.BLACK), padding=inner_container_padding,
                                  width=max_min_container_width - 28)

        self.cont3 = ft.Container(self.column3, alignment=ft.alignment.center, border_radius=border_radius,
                                  border=ft.border.all(2, ft.colors.BLACK), padding=inner_container_padding,
                                  width=max_min_container_width - 28)

        self.cont4 = ft.Container(self.column4, alignment=ft.alignment.center, border_radius=border_radius,
                                  border=ft.border.all(2, ft.colors.BLACK), padding=inner_container_padding,
                                  width=max_min_container_width - 28)
        self.cont5 = ft.Container(self.row10, alignment=ft.alignment.center, border_radius=border_radius,
                                  border=ft.border.all(2, ft.colors.BLACK), padding=inner_container_padding)

    def build(self):
        if self.checkbox_date_start.value:
            for x in self.row_3_controls:
                x.visible = True
        elif not self.checkbox_date_start.value:
            for x in self.row_3_controls:
                x.visible = False
                x.value = ""
        if self.checkbox_date_end.value:
            for x in self.row_5_controls:
                x.visible = True
        elif not self.checkbox_date_end.value:
            for x in self.row_5_controls:
                x.visible = False
                x.value = ""

        if self.checkbox_hour_start.value:
            self.new_task_hour_start.disabled = False
            self.new_task_minute_start.disabled = False
            for x in self.row_7_controls:
                x.visible = True
        elif not self.checkbox_hour_start.value:
            self.new_task_hour_start.disabled = True
            self.new_task_minute_start.disabled = True
            for x in self.row_7_controls:
                x.visible = False
                x.value = ""

        if self.checkbox_hour_end.value:
            self.new_task_hour_end.disabled = False
            self.new_task_minute_end.disabled = False
            for x in self.row_9_controls:
                x.visible = True
        elif not self.checkbox_hour_end.value:
            self.new_task_hour_end.disabled = True
            self.new_task_minute_end.disabled = True
            for x in self.row_9_controls:
                x.visible = False
                x.value = ""

        if self.mode == "one":
            self.new_task_explanation.width = 200
            container_0 = ft.Container(content=self.row0, margin=ft.margin.only(left=fixed_margin, right=fixed_margin))
            container_1 = ft.Container(content=self.row1, margin=ft.margin.only(left=fixed_margin, right=fixed_margin))
            final_column_controls = [container_0, container_1, self.cont1, self.cont2, self.cont3, self.cont4,
                                     self.cont5]
            final_column = ft.Column(controls=final_column_controls)

            container = ft.Container(content=final_column, border=ft.border.all(2, ft.colors.BLACK),
                                     width=1 * max_min_container_width, alignment=ft.alignment.center,
                                     padding=outer_container_padding,
                                     border_radius=ft.border_radius.only(top_right=fixed_margin,
                                                                         bottom_left=fixed_margin))
            return container

        if self.mode == "two":
            self.cont5.width = 580
            self.new_task_explanation.width = 500
            self.new_task_title.width = 1 * max_min_container_width + 100
            row_title = ft.Row(controls=[self.row0, self.row1], alignment=ft.alignment.center,
                               vertical_alignment=ft.CrossAxisAlignment.CENTER)
            container_0 = ft.Container(content=row_title, margin=ft.margin.only(left=fixed_margin, right=fixed_margin))
            row_2_style_1 = ft.Row(controls=[self.cont1, self.cont2], alignment=ft.MainAxisAlignment.SPACE_EVENLY)
            row_2_style_2 = ft.Row(controls=[self.cont3, self.cont4], alignment=ft.MainAxisAlignment.SPACE_EVENLY)
            row_2_style_3 = ft.Row(controls=[self.cont5], alignment=ft.MainAxisAlignment.SPACE_EVENLY)

            final_column_controls = [container_0, row_2_style_1, row_2_style_2, row_2_style_3]
            final_column = ft.Column(controls=final_column_controls)

            container = ft.Container(content=final_column, border=ft.border.all(2, ft.colors.BLACK),
                                     width=2 * max_min_container_width, alignment=ft.alignment.center,
                                     padding=outer_container_padding,
                                     border_radius=ft.border_radius.only(top_right=fixed_margin,
                                                                         bottom_left=fixed_margin))
            return container
        if self.mode == "three":
            self.cont5.width = 580
            self.new_task_explanation.width = 500
            self.new_task_title.width = 2 * max_min_container_width + 100
            row_title = ft.Row(controls=[self.row0, self.row1], alignment=ft.alignment.center,
                               vertical_alignment=ft.CrossAxisAlignment.CENTER)
            container_0 = ft.Container(content=row_title, margin=ft.margin.only(left=fixed_margin, right=fixed_margin))
            row_3_style_1 = ft.Row(controls=[self.cont1, self.cont2, self.cont3],
                                   alignment=ft.MainAxisAlignment.SPACE_EVENLY)
            row_3_style_2 = ft.Row(controls=[self.cont4, self.cont5], alignment=ft.MainAxisAlignment.SPACE_EVENLY)

            final_column_controls = [container_0, row_3_style_1, row_3_style_2]
            final_column = ft.Column(controls=final_column_controls)

            container = ft.Container(content=final_column, border=ft.border.all(2, ft.colors.BLACK),
                                     width=3 * max_min_container_width, alignment=ft.alignment.center,
                                     padding=outer_container_padding,
                                     border_radius=ft.border_radius.only(top_right=fixed_margin,
                                                                         bottom_left=fixed_margin))
            return container
        if self.mode == "four":
            self.cont5.width = 1180
            self.new_task_explanation.width = 1090
            self.new_task_title.width = 3 * max_min_container_width + 100
            row_title = ft.Row(controls=[self.row0, self.row1], alignment=ft.alignment.center,
                               vertical_alignment=ft.CrossAxisAlignment.CENTER)
            container_0 = ft.Container(content=row_title, margin=ft.margin.only(left=fixed_margin, right=fixed_margin))
            row_4_style = ft.Row(controls=[self.cont1, self.cont2, self.cont3, self.cont4],
                                 alignment=ft.MainAxisAlignment.SPACE_EVENLY)
            row_4_style_1 = ft.Row(controls=[self.cont5],
                                   alignment=ft.MainAxisAlignment.SPACE_EVENLY)
            final_column_controls = [container_0, row_4_style, row_4_style_1]
            final_column = ft.Column(controls=final_column_controls)

            container = ft.Container(content=final_column, border=ft.border.all(2, ft.colors.BLACK),
                                     width=4 * max_min_container_width, alignment=ft.alignment.center,
                                     padding=outer_container_padding,
                                     border_radius=ft.border_radius.only(top_right=fixed_margin,
                                                                         bottom_left=fixed_margin))
            return container

    def check_box_checker(self, e):
        if e.control.data == "1":
            if e.control.value:
                for x in self.row_3_controls:
                    x.visible = True
                    x.update()
                return
            else:
                self.new_task_hour_start.disabled = True
                self.new_task_minute_start.disabled = True
                for x in self.row_3_controls:
                    x.visible = False
                    x.value = ""
                    x.update()
                return

        if e.control.data == "2":
            if e.control.value:
                for x in self.row_5_controls:
                    x.visible = True
                    x.update()
                return

            else:
                self.new_task_hour_end.disabled = True
                self.new_task_minute_end.disabled = True
                for x in self.row_5_controls:
                    x.visible = False
                    x.value = ""
                    x.update()
                return

        if e.control.data == "3":
            if e.control.value:
                for x in self.row_7_controls:
                    x.visible = True
                    x.update()
                return

            else:
                for x in self.row_7_controls:
                    x.visible = False
                    x.value = ""
                    x.update()
                return

        if e.control.data == "4":
            if e.control.value:
                for x in self.row_9_controls:
                    x.visible = True
                    x.update()
                return

            else:
                for x in self.row_9_controls:
                    x.visible = False
                    x.value = ""
                    x.update()
                return

    def hour_controller_start(self, e):
        if self.new_task_hour_system_start.value == str(lang_dict["am"][self.lang]):
            self.new_task_hour_start.disabled = False
            self.new_task_minute_start.disabled = False
            self.new_task_hour_start.options = am_hours_list
            self.new_task_hour_start.update()
            self.new_task_minute_start.update()

        if self.new_task_hour_system_start.value == str(lang_dict["pm"][self.lang]):
            self.new_task_hour_start.disabled = False
            self.new_task_minute_start.disabled = False
            self.new_task_hour_start.options = am_hours_list
            self.new_task_hour_start.update()
            self.new_task_minute_start.update()

        if self.new_task_hour_system_start.value == "24":
            self.new_task_hour_start.disabled = False
            self.new_task_minute_start.disabled = False
            self.new_task_hour_start.options = mil_hours_list
            self.new_task_hour_start.update()
            self.new_task_minute_start.update()
        if self.new_task_hour_system_start.value == "":
            self.new_task_hour_start.disabled = True
            self.new_task_minute_start.disabled = True
            self.new_task_hour_start.options = ""
            self.new_task_hour_start.update()
            self.new_task_minute_start.update()

    def hour_controller_end(self, e):
        if self.new_task_hour_system_end.value == str(lang_dict["am"][self.lang]):
            print("end1", self.new_task_hour_system_end.value, str(lang_dict["am"][self.lang]),
                  type(self.new_task_hour_system_end.value))
            self.new_task_hour_end.disabled = False
            self.new_task_minute_end.disabled = False
            self.new_task_hour_end.options = am_hours_list
            self.new_task_hour_end.update()
            self.new_task_minute_end.update()
        if self.new_task_hour_system_end.value == str(lang_dict["pm"][self.lang]):
            print("end2", self.new_task_hour_system_end.value, str(lang_dict["pm"][self.lang]),
                  type(self.new_task_hour_system_end.value))
            self.new_task_hour_end.disabled = False
            self.new_task_minute_end.disabled = False
            self.new_task_hour_end.options = am_hours_list
            self.new_task_hour_end.update()
            self.new_task_minute_end.update()
        if self.new_task_hour_system_end.value == "24":
            print("end3", self.new_task_hour_system_end.value, "24", type(self.new_task_hour_system_end.value))
            self.new_task_hour_end.disabled = False
            self.new_task_minute_end.disabled = False
            self.new_task_hour_end.options = mil_hours_list
            self.new_task_hour_end.update()
            self.new_task_minute_end.update()
        if self.new_task_hour_system_end.value == "":
            self.new_task_hour_end.disabled = True
            self.new_task_minute_end.disabled = True
            self.new_task_hour_end.options = ""
            self.new_task_hour_end.update()
            self.new_task_minute_end.update()


```

**Describe the results you received:**

![ezgif com-gif-maker](https://user-images.githubusercontent.com/33788630/235225798-5612fe2b-5834-4b47-b695-92f85cf836f2.gif)

**Describe the results you expected:**

I expected it to work properly as it used to. 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:

```
Name: flet
Version: 0.6.1
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page: 
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: c:\users\master_pc\pycharmprojects\fletalert\venv\lib\site-packages
Requires: flet-core, httpx, oauthlib, packaging, watchdog, websocket-client, websockets
Required-by:

```

**Operating system**:

Windows 11 pro

**Additional environment details:**
Pycharm
